doc/date.md

```diff
@@ -0,0 +1,61 @@
+# Date
+
+`Napi::Date` class is a representation of the JavaScript `Date` object. The
+`Napi::Date` class inherits its behavior from `Napi::Value` class
+(for more info see [`Napi::Value`](value.md))
+
+## Methods
+
+### Constructor
+
+Creates a new _empty_ instance of a `Napi::Date` object.
+
+```cpp
+Napi::Date();
+```
+
+Returns a new _empty_ JavaScript `Napi::Date`.
+
```

Comment:
Please add the documentation for non-empty constructor maybe something like this: 

### Constructor

Creates a new instance of the `Napi::Date` object.

```cpp
Napi::Date::Date(napi_env env, napi_value value);
```

- `[in] env`: The `napi_env` environment in which to construct the `Napi::Date` object.
- `[in] value`: The `napi_value` which is a handle for a JavaScript `Date`.

Returns a non-empty `Napi::Date` instance.

Reply:
Maybe we should omit the `Returns...` sentence for the ctors, because ctors don't return a value.